specification/ingest/_types/Processors.ts

```diff
@@ -30,47 +30,233 @@ import { Script } from '@_types/Scripting'
+   * @doc_id enrich-processor
+   */
   enrich?: EnrichProcessor
+  /**
+   * Raises an exception.
+   * This is useful for when you expect a pipeline to fail and want to relay a specific message to the requester.
+   * @doc_id fail-processor
+   */
   fail?: FailProcessor
+  /**
+   * Runs an ingest processor on each element of an array or object.
+   * @doc_id foreach-processor
+   */
   foreach?: ForeachProcessor
+  /**
+   * Converts a JSON string into a structured JSON object.
+   * @doc_id json-processor
+   */
   json?: JsonProcessor
+  /**
+   * The `user_agent` processor extracts details from the user agent string a browser sends with its web requests.
+   * This processor adds this information by default under the `user_agent` field.
+   * @doc_id user-agent-processor
+   */
   user_agent?: UserAgentProcessor
+  /**
+   * This processor helps automatically parse messages (or specific event fields) which are of the `foo=bar` variety.
+   * @doc_id kv-processor
+   */
   kv?: KeyValueProcessor
+  /**
+   * The `geoip` processor adds information about the geographical location of an IPv4 or IPv6 address.
+   * @doc_id geoip-processor
+   */
   geoip?: GeoIpProcessor
+  /**
+   * Extracts structured fields out of a single text field within a document.
+   * You choose which field to extract matched fields from, as well as the grok pattern you expect will match.
+   * A grok pattern is like a regular expression that supports aliased expressions that can be reused.
+   * @doc_id grok-processor
+   */
   grok?: GrokProcessor
+  /**
+   * Converts a string field by applying a regular expression and a replacement.
+   * If the field is an array of string, all members of the array will be converted.
+   * If any non-string values are encountered, the processor will throw an exception.
+   * @doc_id gsub-processor
+   */
   gsub?: GsubProcessor
+  /**
+   * Joins each element of an array into a single string using a separator character between each element.
+   * Throws an error when the field is not an array.
+   * @doc_id join-processor
+   */
   join?: JoinProcessor
+  /**
+   * Converts a string to its lowercase equivalent.
+   * If the field is an array of strings, all members of the array will be converted.
+   * @doc_id lowercase-processor
+   */
   lowercase?: LowercaseProcessor
+  /**
+   * Removes existing fields.
+   * If one field doesn’t exist, an exception will be thrown.
+   * @doc_id remove-processor
+   */
   remove?: RemoveProcessor
+  /**
+   * Renames an existing field.
+   * If the field doesn’t exist or the new name is already used, an exception will be thrown.
+   * @doc_id rename-processor
+   */
   rename?: RenameProcessor
+  /**
+   * Runs an inline or stored script on incoming documents.
+   * The script runs in the `ingest` context.
+   * @doc_id script-processor
+   */
   script?: Script
+  /**
+   * Sets one field and associates it with the specified value.
```

Comment:
```suggestion
   * Adds a field with the specified value.
```

Reply:
I know this is the original description, but I find it a bit cryptic. How about this simpler one?